src/lib/hooks/use-wallet.ts

```diff
@@ -1,35 +1,46 @@
 import { providers } from 'ethers'
-import { useEffect, useState } from 'react'
-import { Hex } from 'viem'
-import { useAccount, useNetwork, useWalletClient } from 'wagmi'
+import { Hex, HttpTransport, PublicClient } from 'viem'
+import { useAccount, usePublicClient } from 'wagmi'
 
-import { getEthersProvider, getEthersSigner } from '../utils/ethers-adapters'
+import { useNetwork } from '@/lib/hooks/use-network'
+
+export interface IndexRpcProvider {
+  estimateGas(tx: any): Promise<bigint>
+  getGasPrice(): Promise<bigint>
+}
 
 type Account = {
   address: Hex | undefined
-  provider: any | undefined
-  signer: any | undefined
```

Comment:
The signer wasn't really needed for most use cases - only gas estimation used it and now works with the provider using that change of always adding `account` to the tx object